torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -231,6 +231,15 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
       return c10::optional<at::Tensor>(updated_val);
     }
     return c10::nullopt;
+  } else if (node->kind() == onnx::Reshape) {
+    assert(inputTensorValues.size() == 2);
+    updated_val = inputTensorValues[0];
+    std::vector<int64_t> shape(inputTensorValues[1].sizes()[0], 0);
+    auto shape_a = inputTensorValues[1].accessor<int64_t, 1>();
+    for (size_t i = 0; i < inputTensorValues[1].sizes()[0]; ++i) {
+      shape[i] = shape_a[i];
```

Comment:
Since we are dealing with an ONNX graph, it is possible that the reshape node's second input has `-1` or `0` in the new shape. The code here needs to handle those situations, before passing it along to ATen op.

Reply:
Thanks for pointing this out.
-1 dim value is actually supported in aten. This is actually tested with mask rcnn model.
But for 0 values, I updated the PR to address this.